packages/stories/stories-the-dot-banks/src/toolbox/ScrollyFrame.tsx

```diff
@@ -0,0 +1,128 @@
+// TODO: glamor?
+import { css, merge } from 'glamor'
+import React, { useEffect, useRef, useState } from 'react'
+
+// TODO: get rid of styleguide dep
+import { useColorContext } from '../__styleguide/components/Colors/ColorContext'
+import * as mediaQueries from '../__styleguide/theme/mediaQueries'
+
+import { StepsMenu } from './StepsMenu'
+
+export type StoryGraphicProps = {
+  step: number
+  activeColor?: string
+}
+
+export type StoryGraphicType = React.FC<StoryGraphicProps>
+
+/*
+  This component provides the frame around the graphical element
+  of the scrolly. It also calculates the current step, as well as
+  listens to the active color (if applicable) and passes both to
+  the graphical element.
+ */
+export const ScrollyFrame = ({
+  stepIds,
+  Graphic,
+}: {
+  stepIds: string[]
+  Graphic: StoryGraphicType
+}) => {
+  // TODO: migrate color ctxt out of styleguide
+  const [colorScheme] = useColorContext()
+
+  // TODO: migrate header height ctxt out of styleguide
+  // const [headerHeight] = useHeaderHeight()
+  const headerHeight = 80
+
+  const [currentStep, setCurrentStep] = useState<number>(1)
+  const [activeColor, setActiveColor] = useState<string>(undefined)
+  const [isFixed, setFixed] = useState<boolean>(false)
+
+  const containerRef = useRef<HTMLDivElement>()
+  const chartRef = useRef<HTMLDivElement>()
+
+  // TODO: some error handling would be good
+  // TODO: refine the scroll/steps logic
+  const handleScroll = () => {
```

Comment:
This contains a lot of DOM measuring with `getBoundingClientRect`. For performance reasons we should use IntersectionObserver instead.